plugins/modules/pacman_key.py

```diff
@@ -146,47 +195,71 @@ def __init__(self, module):
         force_update = module.params['force_update']
         keyring = module.params['keyring']
         state = module.params['state']
+        ensure_trusted = module.params['ensure_trusted']
         self.keylength = 40
 
         # sanitise key ID & check if key exists in the keyring
         keyid = self.sanitise_keyid(keyid)
-        key_present = self.key_in_keyring(keyring, keyid)
+        key_validity = self.key_validity(keyring, keyid)
+        key_present = len(key_validity) > 0
+        key_valid = any(key_validity)
 
         # check mode
         if module.check_mode:
-            if state == "present":
+            if state == 'present':
                 changed = (key_present and force_update) or not key_present
+                if not changed and ensure_trusted:
+                    changed = not (key_valid and self.key_is_trusted(keyring, keyid))
                 module.exit_json(changed=changed)
-            elif state == "absent":
-                if key_present:
-                    module.exit_json(changed=True)
-                module.exit_json(changed=False)
+            if state == 'absent':
+                module.exit_json(changed=key_present)
 
-        if state == "present":
-            if key_present and not force_update:
+        if state == 'present':
+            trusted = key_valid and self.key_is_trusted(keyring, keyid)
+            if not force_update and key_present and (not ensure_trusted or trusted):
                 module.exit_json(changed=False)
-
+            changed = False
             if data:
                 file = self.save_key(data)
                 self.add_key(keyring, file, keyid, verify)
-                module.exit_json(changed=True)
+                changed = True
             elif file:
                 self.add_key(keyring, file, keyid, verify)
-                module.exit_json(changed=True)
+                changed = True
             elif url:
                 data = self.fetch_key(url)
                 file = self.save_key(data)
                 self.add_key(keyring, file, keyid, verify)
-                module.exit_json(changed=True)
+                changed = True
             elif keyserver:
                 self.recv_key(keyring, keyid, keyserver)
-                module.exit_json(changed=True)
-        elif state == "absent":
+                changed = True
+            if changed or (ensure_trusted and not trusted):
+                self.lsign_key(keyring=keyring, keyid=keyid)
+                changed = True
+            module.exit_json(changed=changed)
+        elif state == 'absent':
             if key_present:
                 self.remove_key(keyring, keyid)
                 module.exit_json(changed=True)
             module.exit_json(changed=False)
 
+    def gpg(self, args, keyring=None, **kwargs):
```

Comment:
Apparently `**kwargs` is here just to handle the different cases of  `check_rc`. That being the case, I would suggest to drop the `**` thing and make it explicit:
```suggestion
    def gpg(self, args, keyring=None, check_rc=True):
```
or whichever default value you prefer, then adjust the calls to that method accordingly - I mean, the calls will work as they are right now, but you might want to remove the extra parameter for the default value (`True` in my suggestion).

Reply:
The function doesn't do anything with `check_rc`. IMHO `**kwargs` conveys that it's only passed through.

Reply:
@russoz means that `kwargs` right now is only ever used for `check_rc`, so it's better to make this more explicit.

The same is true for `pacman_key()`'s `**kwargs`.

Reply:
Not that strong. I still think the readability would be improved by naming - or removing that parameter as it is not used  - but happy to merge the PR as-is wrt this.